src/cosmic_ray/operators/util.py

```diff
@@ -19,3 +19,134 @@ def dec(cls):
         return cls
 
     return dec
+
+
+def dump_node(node, stdout=None):
+    import sys
+    stdout = stdout or sys.stdout
+    write = stdout.write
+
+    def do_dump(node, indent=''):
+        write(f"{indent}{type(node).__name__}({node.type}")
+        value = getattr(node, 'value', None)
+        if value:
+            value = value.replace('\n', '\\n')
+            write(f", '{value}'")
+        children = getattr(node, 'children', None)
+        if children:
+            write(', [\n')
+            for child in children:
+                do_dump(child, indent+' '*4)
+                write(',\n')
+            write(f"{indent}]")
+        write(')')
+        if not indent:
+            write('\n')
+
+    do_dump(node)
+
+
+class ObjTest:
```

Comment:
I think this `ObjTest` system is very interesting and potentially very useful for other operators. If for some reason we don't accept this PR, could you submit this separately? It would be a shame to lose it.